leptonai/cli/photon.py

```diff
@@ -635,268 +407,198 @@ def run(
-                    memory=replica_memory,
-                    accelerator_type=replica_accelerator_type,
-                    accelerator_num=replica_accelerator_num,
-                    ephemeral_storage_in_gb=replica_ephemeral_storage_in_gb,
-                    affinity=(
-                        LeptonResourceAffinity(
-                            allowed_dedicated_node_groups=[resource_affinity]
-                        )
-                        if resource_affinity
-                        else None
-                    ),
-                    min_replicas=min_replicas,
-                    max_replicas=max_replicas,
-                ),
-                mounts=make_mounts_from_strings(mounts),
-                image_pull_secrets=image_pull_secrets,
-                envs=make_env_vars_from_strings(env_list, secret_list),
-                api_tokens=make_token_vars_from_config(public, tokens),
-                auto_scaler=AutoScaler(
-                    scale_down=(
-                        ScaleDown(no_traffic_timeout=no_traffic_timeout)
-                        if no_traffic_timeout
-                        else None
-                    ),
-                    target_gpu_utilization_percentage=target_gpu_utilization,
-                ),
-                health=HealthCheck(
-                    liveness=(
-                        HealthCheckLiveness(initial_delay_seconds=initial_delay_seconds)
-                        if initial_delay_seconds
-                        else None
-                    )
-                ),
-            )
 
-            lepton_deployment = LeptonDeployment(
-                metadata=Metadata(id=id, name=deployment_name), spec=spec
-            )
-            client.deployment.create(lepton_deployment)
-            console.print(
-                f"Photon launched as [green]{deployment_name}[/]. Use `lep deployment"
-                f" status -n {deployment_name}` to check the status."
-            )
-        except ValueError as e:
+@photon.command()
+@click.option("--name", "-n", type=str, help="Name of the photon to run.")
+@click.option(
+    "--model",
+    "-m",
+    type=str,
+    help=(
+        "Model spec of the photon. If model is specified, we will rebuild the photon"
+        " before running."
+    ),
+)
+@click.option(
+    "--file", "-f", "path", type=str, help="Path to the specific `.photon` file to run."
+)
+@click.option(
+    "--port", "-p", type=int, help="Port to run on.", default=config.DEFAULT_PORT
+)
+@click.option(
+    "--env",
+    "-e",
+    help="Environment variables to pass to the deployment, in the format `NAME=VALUE`.",
+    multiple=True,
+)
+@click.option(
+    "--secret",
+    "-s",
+    help=(
+        "Secrets to pass to the deployment, in the format `NAME=SECRET_NAME`. If"
+        " secret name is also the environment variable name, you can"
+        " omit it and simply pass `SECRET_NAME`."
+    ),
+    multiple=True,
+)
+@click.pass_context
+def run_local(
```

Comment:
name it runlocal? (I think right now it works because we do fuzzy match)

Reply:
Ah dang, nice catch, yeah.